pkg/agent/endpoints/endpoints.go

```diff
@@ -37,6 +37,13 @@ func (e *endpoints) Start() error {
 		return err
 	}
 
+	if e.c.GRPCHook != nil {
+		err = e.c.GRPCHook(e.grpc)
```

Comment:
Could having access to the endpoints.Config be useful for the extender of the server? I mean maybe we should also pass e.c to GRPCHook().
Idea: In addition to this I think it would be great if we could read/parse custom configuration from spire-agent and spire-server config files to allow the extender to leverage such capability.

Reply:
>Could having access to the endpoints.Config be useful for the extender of the server?
I put a little bit of thought into some use cases for this, but couldn't come up with many since the consumer configures this hook in the Config itself, and therefor should have ready access to those values when the hook is declared... I also would be weary of someone using the hook to modify the active config or do something else that we didn't count on... if you can think of a good use case though I'm happy to explore this further

>In addition to this I think it would be great if we could read/parse custom configuration from spire-agent and spire-server config files
Trying to understand what you mean here - can you provide an example?

Reply:
Well, the "Idea" only has meaning if we pass the configuration to the hook. I imagined that the user could write a configuration like:
```
server {
    bind_address = "127.0.0.1"
    bind_port = "8081"
    ...
    extensions {
        ext_name_1 {
            prop1: value1
            prop2: value2
        }
    }
}
```
Then from the hook the extension would have access to the "ext_name_1" configuration (actually it would have access to all the configuration if we pass the whole config instance, but ideally it only cares for that piece of config). I think this only is useful to someone that wants to expose new endpoints and needs some configuration for it, but he/she is too lazy to make its own configuration reader...

Reply:
Also, we can pass an inmutable copy of the config if modifying the values is a major concern.

Reply:
Ah I see... In order to do that though, we'd have to pass the agent/server config (or at least that bit) into the Endpoints config, and that feels like overkill... As for immutability, I don't think golang supports that. I feel it is cleaner to leave grpc.Server as the only arg, since that is the intended target for the hook... folks that want to access other config information from inside the hook can do something like this:
```go
c := &endpoints.Config{
     BindAddr: foo,
     GRPCHook: func(s *grpc.Server) error { return myHook(s, foo) },
}
```
We can change the signature in the future if we need it? But I don't think it would prevent things from being done, it would be more of a convenience if anything

Reply:
Makes sense, if we need something like this in the future then we can consider working on this.